trade-finance-manager-api/src/v1/repo/user.repo.update-user-by-id.test.ts

```diff
@@ -0,0 +1,121 @@
+    });
+  });
+
+  describe('update user by id', () => {
+    let updateUserRequest: UpdateUserRequest;
+    let updateUserDatabaseRequest: UpdateUserRequest & { auditRecord: AuditDatabaseRecord };
+    let updateUserDatabaseResponse: WithId<TfmUser>;
+    let auditDetails: AuditDetails;
+    let auditRecord: AuditDatabaseRecord;
+    let userId: ObjectId;
+
+    beforeAll(() => {
+      jest.useFakeTimers();
+    });
+
+    beforeEach(() => {
+      auditDetails = generateSystemAuditDetails();
+      auditRecord = generateAuditDatabaseRecordFromAuditDetails(auditDetails);
+
+      updateUserRequest = aValidUpsertUserRequest();
+      updateUserDatabaseRequest = { ...updateUserRequest, auditRecord };
+      updateUserDatabaseResponse = { ...updateUserDatabaseRequest, _id: userId, status: USER.STATUS.ACTIVE } as WithId<TfmUser>;
+
+      userId = new ObjectId();
+    });
+
+    afterAll(() => {
+      jest.useRealTimers();
+    });
+
+    describe('when the update is successful', () => {
+      beforeEach(() => {
+        mockSuccessfulFindOneAndUpdateResponse();
+      });
+
+      it('calls the tfm user collection', async () => {
+        await makeRequest();
+
+        expect(getCollectionMock).toHaveBeenCalledWith(MONGO_DB_COLLECTIONS.TFM_USERS);
+        expect(getCollectionMock).toHaveBeenCalledTimes(1);
+      });
+
+      it('updates a user', async () => {
+        const filter = { _id: { $eq: userId } };
+        const update = {
+          $set: {
+            ...updateUserDatabaseRequest,
+            auditRecord: generateAuditDatabaseRecordFromAuditDetails(auditDetails),
+          },
+        };
+        const options: FindOneAndUpdateOptions = { returnDocument: 'after' };
+
+        await makeRequest();
+
+        expect(findOneAndUpdateMock).toHaveBeenCalledWith(filter, update, options);
+        expect(findOneAndUpdateMock).toHaveBeenCalledTimes(1);
+      });
+
+      it('returns the updated user', async () => {
+        const result = await makeRequest();
+
+        expect(result).toEqual(updateUserDatabaseResponse);
+      });
+    });
+
+    describe('when the update is unsuccessful', () => {
+      beforeEach(() => {
+        mockUnsuccessfulFindOneAndUpdateResponse();
+      });
+
+      it('throws an error', async () => {
+        await expect(makeRequest()).rejects.toThrow(DocumentNotUpdatedError);
+      });
+    });
+
+    function mockFindOneAndUpdateResponse(response: ModifyResult) {
+      findOneAndUpdateMock = jest.fn(() => response);
+      getCollectionMock = jest.fn().mockResolvedValue({
+        ...getCollectionMock,
+        findOneAndUpdate: findOneAndUpdateMock,
+      });
+
+      jest.spyOn(mongoDbClient, 'getCollection').mockImplementation(getCollectionMock);
+    }
+
+    function mockSuccessfulFindOneAndUpdateResponse() {
+      mockFindOneAndUpdateResponse({ ok: 1, value: updateUserDatabaseResponse });
+    }
+
+    function mockUnsuccessfulFindOneAndUpdateResponse() {
```

Comment:
Similar to previous comments, `mockSuccessfulFindOneAndUpdateResponse` and `mockUnsuccessfulFindOneAndUpdateResponse` are only called once in individual before each's, I don't see a need to abstract into another function

Reply:
This was to allow the tests to be read easier -- I'll remove these